app/uk/gov/hmrc/gform/validation/ComponentValidator.scala

```diff
@@ -266,10 +306,22 @@ object ComponentValidator {
     messages: Messages,
     sse: SmartStringEvaluator
   ) = {
+    val valueWithoutInvisibleCharacters =
+      value.filterNot(skipCharactersForValidations contains _)
+
     val ValidText =
       """[A-Za-z0-9\(\)\,\'\’\“\”\%\•\-\.\r\s\£\\n\+\;\:\*\?\=\/\&\!\@\#\$\€\`\~\"\<\>\_\§\±\[\]\{\}\–\—\‘\’\“\”]+""".r
-    val messageKey = "generic.longText.error.pattern"
-    sharedTextComponentValidator(fieldValue, value, min, max, ValidText, messageKey)
+
+    invalidCharactersValidator(fieldValue, valueWithoutInvisibleCharacters, ValidText, genericLongTextErrorPattern)
```

Comment:
We need to do same validation for `shortText` and various `numbers` fields. Actually I think safe thing would be to do this for any input we are receiving from the user.